gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/step/filter/RangeGlobalStep.java

```diff
@@ -129,6 +170,9 @@ public boolean equals(final Object other) {
 
     @Override
     public Set<TraverserRequirement> getRequirements() {
+        if (this.getTraversal().getParent() instanceof RepeatStep) {
```

Comment:
The tests all seem to pass with this code removed (I only ran tests through tinkergraph-gremlin but that should be sufficient). The getRequirements() method in RepeatStep should already take care of these cases, as the presence of a single RepeatStep in a traversal will add a `SINGLE_LOOP` requirement, and if a RepeatStep wraps a child which itself contains a repeat, it will add `NESTED_LOOP`. That alone should guarantee these requirements will be present anytime it is needed for this step.